src/beatree/mod.rs

```diff
@@ -265,6 +265,7 @@ impl Tree {
                 staged_changeset.clone(),
                 &mut bbn_index,
                 &mut branch_node_pool,
+                todo!(), // leaf store reader
```

Comment:
god dammit. This would require a bit of rejigging.

cc @gabriele-0201

Reply:
Yes, this will definitely require a little bit of rejuggling, but I don't think it is as painful as it seems. The main problem I can see here is that the `LeafStoreReader` is under the shared lock, and we don't want to lock out lookups during the sync phase. I see three approaches:
1. We find a way to share the `LeafStoreReader` between the `shared` and `sync` locks (hardest)
2. We just create two separate `LeafStoreReader`s, one for `sync` and one for `shared` (easiest)
3. We could just give read ability to the `LeafStoreWriter`

Interestingly, `LeafStoreWriter` is currently not really a writer, it is just a cache of changes that will be applied during the `sync` phase

Anyway, #54 contains the implementation of the second proposed solution!

Reply:
it's unavoidable - we have to read leaves during update!

Yes, it's kind of annoying because currently the leaf store reader is only under the shared lock. Two instances of the leaf store reader seems fine to me, but I'm not as familiar with that area of the code